lib/HttpClient.ts

```diff
@@ -226,63 +247,93 @@ export class HttpClient {
         this._disposed = true;
     }
 
-    private _requestRaw(info: RequestInfo, data: string | NodeJS.ReadableStream): Promise<HttpClientResponse> {
-        return new Promise<HttpClientResponse>((resolve, reject) => {
-            let socket;
+    /**
+     * Raw request.
+     * @param info 
+     * @param data 
+     */
+    public requestRaw(info: ifm.IRequestInfo, data: string | NodeJS.ReadableStream): Promise<ifm.IHttpClientResponse> {
```

Comment:
do we need to expose these publicly? can't users turn off redirects and not pass auth handlers to get the same behavior? extra complexity to have multiple ways to achieve the same goal.

Reply:
We do give them the request method which is relatively low level. We need requestRawWithCallbacks for the NTLM auth handler to use but we may not need this one. I would be OK putting this one back to private and only having the version with callbacks be public for the auth handlers and others can use it as their low level request if need be.

@bryanmacfarlane Does that work for you?